### PR TITLE
Make arguments optional

### LINKED ISSUE #63: Test optional arguments
- [x] None
- [x] Name, opts
- [x] Name, opts, callback
- [x] Name, callback
- [x] Callback

### PR DESCRIPTION
**Pull request by vweevers**
Closes #63.

### CODE REVIEW

**Comment by vweevers** [comment]
Hmm should it also support:

```js
const level = require('level-test')({ mem: true })
const db = level({ valueEncoding: 'json' })
```

As the name / location is irrelevant for `memdown`, and defaults to a random name for others.

File: browser.js [hunk 1]
```diff
@@ -9,6 +9,19 @@ function wrap (down, parentOpts) {
...
+      loc = null
+      opts = null
+    } else if (typeof loc === 'object' && loc !== null) { // (opts[, cb])
+      cb = opts
+      opts = loc
+      loc = null
+    } else if (typeof opts === 'function') { // (name, cb)
+      cb = opts
+      opts = null
+    }
```
Comment (ralphtheninja) [inline browser.js]: Can we somehow make this into a function? I was thinking we could do something like

```js
[ cb, loc, opts ] = argify(loc, opts, cb)
```

Comment (vweevers) [reply browser.js]: We don't support node 4 here, so that could work.

Comment (ralphtheninja) [reply browser.js]: Well, we currently support node 4.

Comment (vweevers) [reply browser.js]: Oh, it's not in travis.

Comment (vweevers) [reply browser.js]: #68

**Pull request merged**